tinygrad/codegen/optimizer.py

```diff
@@ -282,6 +282,33 @@ def hand_coded_optimizations(self):
         # early exit
         return
 
+    # should use matvec
+    if getenv("MV",1) != 0 and self.opts.has_local and \
+        self.reduceop and self.reduceop.op == ReduceOps.SUM and \
+        isinstance(self.reduceop.src[0], LazyOp) and self.reduceop.src[0].op == BinaryOps.MUL and \
+        isinstance(self.reduceop.src[0].src[0], LazyBuffer) and isinstance(self.reduceop.src[0].src[1], LazyBuffer) and \
+        len(self.full_shape) >= 2:
```

Comment:
Is it faster to first check the len before all of those (expensive) isinstance calls?

Reply:
Fixed!